_episodes/01-connecting.md

```diff
@@ -129,6 +128,93 @@ For those logging in with PuTTY it would likely be best to cover the terminal
 basics already mentioned above before moving on to navigating the remote
 system.
 
+## Creating an SSH key
+
+SSH keys are an alternative method for authentication to obtain access to 
+remote computing systems. They can also be used for authentication when
+transferring files or for accessing version control systems. In this section
+you will create a pair of SSH keys, a private key which you keep on your
+own computer and a public key which is placed on the remote HPC system
+that you will log in to.
+
+### Linux, Mac and Windows Subsystem for Linux
+
+Once you have opened a terminal check for existing SSH keys and filenames
+since existing SSH keys are overwritten,
+```
+$ ls ~/.ssh/
+```
+{: .language-bash}
+
+then generate a new public-private key pair,
+```
+$ ssh-keygen -t ed25519 -a 100 -f ~/.ssh/id_{{ site.workshop_host }}_ed25519
+```
+{: .language-bash}
+
+- `-o` (no default): use the OpenSSH key format,
+  rather than PEM.
+- `-a` (default is 16): number of rounds of passphrase derivation;
+  increase to slow down brute force attacks.
+- `-t` (default is [rsa](https://en.wikipedia.org/wiki/RSA_(cryptosystem))):
+  specify the "type" or cryptographic algorithm. 
+  [ed25519](https://en.wikipedia.org/wiki/EdDSA)
+ is faster and shorter than RSA for comparable strength.
+- `-f` (default is /home/user/.ssh/id_algorithm): filename to store your keys.
+  If you already have SSH keys, make sure you specify a different name:
+  `ssh-keygen` will overwrite the default key if you don't specify!
+
+If EcDSA is not available, use the older (but strong and trusted)
+[RSA](https://en.wikipedia.org/wiki/RSA_(cryptosystem)) cryptography:
+
+```
+$ ls ~/.ssh/
+$ ssh-keygen -o -a 100 -t rsa -b 4096 -f ~/.ssh/id_{{ site.workshop_host }}_rsa
+```
+{: .language-bash}
+ 
+The flag `-b` sets the number of bits in the key.
+The default is 2048. EdDSA uses a fixed key length,
+so this flag would have no effect.
+
+When prompted, enter a strong password that you will remember. 
+Cryptography is only as good as the weakest link, and this will be 
+used to connect to a powerful, precious, computational resource.
+
+Take a look in `~/.ssh` (use `ls ~/.ssh`). You should see the two 
+new files: your private key (`~/.ssh/key_{{ site.workshop_host }}_ed25519` 
+or `~/.ssh/key_{{ site.workshop_host }}_rsa`) and 
+the public key (`~/.ssh/key_{{ site.workshop_host }}_ed25519.pub` or
+`~/.ssh/key_{{ site.workshop_host }}_rsa.pub`). If a key is 
+requested by the system administrators, the *public* key is the one
+to provide.
+
+> ##### Private keys are your private identity
+>
+> A private key that is visible to anyone but you should be considered compromised,
```

Comment:
This section runs wider than 79 characters, producing a fragmented diff.

Reply:
Thanks. Fixed.